### DISCUSSION TITLE
ReDrop

### DISCUSSION THREAD

**Discussion by ceebeel** [projectM-visualizer/projectm#821] [Show and tell]
Hi,

I'm currently implementing ProjectM in Rust using the `egui` and `cpal` libraries.
For the moment the project is in alpha version: 
- There are a few (known) bugs to fix.
- The audio input is the defined by your system (default). (You can't choose).
- Finaly, before first release, the codebase need to be refactored.

All the libraries are cross-platform, but I've only tested them on Windows and Linux.

All feedbacks are welcome !

Enjoy [ReDrop](https://github.com/ceebeel/ReDrop) ! 🤪

CeeBeeL

**Comment by blaquewithaq** [comment]
Glad to see someone making use of those rust bindings! :P

**Reply by ceebeel** [reply]
Hehe, Thank you for doing it! 😉

**Comment by revmischa** [comment]
Okay that's amazing, thank you. We can move the project under the projectM-visualizer org if you wish.
Also please be aware I was working on a rust frontend as well with SDL. I start with SDL2 then started trying to make sure of SDL3, which ended up in me creating rust SDL3 bindings as well. Maybe that will be of some use to you or we can combine those efforts or something.

https://github.com/projectM-visualizer/frontend-sdl-rust

Always happy to chat on the projectM discord

**Reply by ceebeel** [reply]
Nice ! 
My code is largely inspired by yours.
Where can I join the chat?

**Reply by revmischa** [reply]
https://discord.gg/mMrxAqaa3W
